test/test.downscopedclient.ts

```diff
@@ -534,12 +536,100 @@ describe('DownscopedClient', () => {
     it('should throw when the source AuthClient rejects on token request', async () => {
       const expectedError = new Error('Cannot get subject token.');
       client.throwError = true;
+
       const downscopedClient = new DownscopedClient(
         client,
         testClientAccessBoundary
       );
       await assert.rejects(downscopedClient.getAccessToken(), expectedError);
     });
+
+    it('should copy source cred expiry time if STS response does not return expiry time', async () => {
+      const now = new Date().getTime();
+      const credentials = {
+        access_token: 'ACCESS_TOKEN',
+        expiry_date: now + ONE_HOUR_IN_SECS * 1000,
+      };
+      const stsSuccessfulResponseWithoutExpireInField = Object.assign(
+        stsSuccessfulResponse,
+        {}
+      );
+      delete stsSuccessfulResponseWithoutExpireInField.expires_in;
+      const scope = mockStsTokenExchange([
+        {
+          statusCode: 200,
+          response: stsSuccessfulResponseWithoutExpireInField,
+          request: {
+            grant_type: 'urn:ietf:params:oauth:grant-type:token-exchange',
+            requested_token_type:
+              'urn:ietf:params:oauth:token-type:access_token',
+            subject_token: 'subject_token_0',
+            subject_token_type: 'urn:ietf:params:oauth:token-type:access_token',
+            options:
+              testClientAccessBoundary &&
+              JSON.stringify(testClientAccessBoundary),
+          },
+        },
+      ]);
+
+      client.setCredentials(credentials);
+      const downscopedClient = new DownscopedClient(
+        client,
+        testClientAccessBoundary
+      );
+
+      const tokenResponse = await downscopedClient.getAccessToken();
+      assert.deepStrictEqual(
+        tokenResponse.token,
+        stsSuccessfulResponseWithoutExpireInField.access_token
+      );
+      assert.strictEqual(
+        downscopedClient.credentials.expiry_date,
+        credentials.expiry_date
+      );
+      scope.done();
+    });
+
+    it('should have no expiry date if source cred has no expiry time and STS response does not return one', async () => {
+      const credentials = {
+        access_token: 'ACCESS_TOKEN',
+      };
```

Comment:
I think `TestAuthClient ` already doesn't populate the expiration, so this may not be needed.